test/cpp/naming/resolver_component_test_shell_script_wrapper.cc

```diff
@@ -0,0 +1,55 @@
+/*
+ *
+ * Copyright 2017 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <grpc/grpc.h>
+#include <grpc/support/alloc.h>
+#include <grpc/support/log.h>
+#include <grpc/support/string_util.h>
+#include <unistd.h>
+
+#include <string>
+#include <vector>
+
+#include "test/cpp/util/test_config.h"
+
+extern "C" {
+#include "test/core/util/port.h"
+}
+
+#define NUM_ARGS 4
+
+int main(int argc, char **argv) {
+  grpc::testing::InitTest(&argc, &argv, true);
+  grpc_init();
+  char *exec_args[NUM_ARGS];
+  exec_args[0] = gpr_strdup(
```

Comment:
There's probably no need for `gpr_strdup()` for any of these args.  The values look like they will all continue to exist until after `execv()` is called, and it's kind of messy to allocate them and not free them later.